src/main/java/com/cloudbees/jenkins/GitHubPushTrigger.java

```diff
@@ -155,7 +162,11 @@ public void run() {
      * Returns the file that records the last/current polling activity.
      */
     public File getLogFile() {
-        return new File(job.getRootDir(), "github-polling.log");
+        if (job == null) {
+            return null;
```

Comment:
not sure this is a good idea, but it does not make the situation worse at least. It would have been NPE anyway.
Would be nice to update Javadoc to document the new behavior